pkg/apis/kubermatic/v1/settings.go

```diff
@@ -91,6 +91,9 @@ type SettingSpec struct {
 	// MachineDeploymentVMResourceQuota is used to filter out allowed machine flavors based on the specified resource limits like CPU, Memory, and GPU etc.
 	MachineDeploymentVMResourceQuota *MachineFlavorFilter `json:"machineDeploymentVMResourceQuota,omitempty"`
 
+	// AvailableOS shows the available operating systems to use in the machine deployment.
+	AvailableOperatingSystem AvailableOS `json:"availableOperatingSystem,omitempty"`
+
```

Comment:
Instead of a struct like this let's go for a map 
```go
// allowedOperatingSystems defines a map of operating systems that can be used for the machines.
type allowedOperatingSystems map[providerconfig.OperatingSystem]bool

type SettingSpec struct {
   ...
   // Allowed operating systems to use on the dashboard. Leave empty to use the default supported operating systems for the cloud providers.
   // +optional
   AllowedOperatingSystems allowedOperatingSystems `json:"allowedOperatingSystems,omitempty"`
   ...
}
```